docs/developer/hardware.md

```diff
@@ -0,0 +1,163 @@
+# Hardware
+
+Many hardware targets will work as a physical Arcade platform. These are typically SoC based systems that run a small core operating system.
+
+## Hardware definitions
+
+Arcade hardware targets are supported by their _hardware definition_ which is provided as an [extension](https://makecode.com/extensions). The hardware definition specifies the base device characteristics and mappings for the screen, buttons, pins, and sound.
+
+The components of the extension are source files which specify device mappings and control behaviors that are specific to the hardware target. The files and dependancies are all declared in a ``pxt.json`` file just like all MakeCode [extensions](https://makecode.com/extensions).
+
+```
+{
+    "name": "gizmoq",
+    "description": "GizmoQ board",
+    "files": [
+        "config.ts"
+    ],
+    "card": {
+        "name": "GizmoQ Gamer Board",
+        "description": "Specialized GizmoQ Linux image for the GizmoQ Board.",
+        "learnMoreUrl": "https://arcade.makecode.com/boards/gizmoq",
+        "buyUrl": "https://www.bing.com/search?q=gizmoq+board",
+        "cardType": "hw",
+        "imageUrl": "/static/boards/gizmoq.jpg"
+    },
+    "dependencies": {
+        "core": "file:../core---gizmoq",
+        "screen": "file:../screen---gizmoq",
+        "mixer": "file:../mixer---gizmoq",
+        "game": "file:../game"
+    },
+    "public": true
+}
+```
+
+When downloading an Arcade game program, the **GizmoQ** hardware will appear as a card selection in the **Choose your Hardware** dialog. The board choice might look something like this:
+
+```codecard
+[{
+        "name": "GizmoQ Gamer Board",
+        "description": "Specialized GizmoQ Linux image for the GizmoQ Board.",
+        "learnMoreUrl": "https://arcade.makecode.com/boards/gizmoq",
+        "buyUrl": "https://www.bing.com/search?q=gizmoq+board",
+        "cardType": "hw",
+        "imageUrl": "/static/boards/gizmoq.png"
+}]
+```
+
+### Config description
+
+The hardware definition extension contains a ``config.ts`` file which exports constants that define device settings for your board. These are mostly pin an button settings which map to DAL defines. You get a ``dal.d.ts`` generated for you in the ``core`` extension which contains the DAL definitions. The ``config.ts`` also has the screen settings for board. Here's a sample ``config.ts`` file:
```

Comment:
@mmoskal  has a nifty web page to sniff and update the pin definition in the bootloader

Reply:
Nice. Eventually we'll compile all the elements of the board dev 'ecosystem' and put them in here.